serialization_frozen_test.go

```diff
@@ -126,3 +131,90 @@ func TestFrozenFormat(t *testing.T) {
 		})
 	}
 }
+
+type frozenTestCase struct {
+	name        string
+	useMMapped  bool
+	useFrozen   bool
+	shouldPanic bool
+}
+
+func TestFrozenCases(t *testing.T) {
+	cases := []frozenTestCase{
+		{name: "in-memory-frozen", useMMapped: false,
+			useFrozen:   true,
+			shouldPanic: false},
+		{name: "mmap-frozen", useMMapped: true,
+			useFrozen: true,
+			// THIS SHOULD NOT BE PANIC/FAULTING
+			shouldPanic: true},
+		{name: "in-memory", useMMapped: false,
+			useFrozen:   false,
+			shouldPanic: false},
+		{name: "mmap", useMMapped: true,
+			useFrozen:   false,
+			shouldPanic: false},
+	}
+	for _, testCase := range cases {
+		testFrozenCase(t, testCase)
+	}
+}
+func testFrozenCase(t *testing.T, testCase frozenTestCase) {
+
+	startingBitmap := NewBitmap()
+	startingBitmap.highlowcontainer.appendContainer(0, &runContainer16{iv: []interval16{{0, 0},
+		{2, 5}}}, false)
+
+	primary := getBitmap(t, testCase, startingBitmap)
+	assert.True(t, startingBitmap.Equals(primary))
+	clone := primary.Clone()
+	primary.Xor(clone)
+	res := primary.GetCardinality()
+	assert.Equal(t, res, uint64(0))
+	if testCase.shouldPanic {
+		assert.Panics(t, func() {
+			debug.SetPanicOnFault(true)
+			primary.Xor(clone)
+		})
+	} else {
+		primary.Xor(clone)
+	}
+}
+
+func getBitmap(t *testing.T, testCase frozenTestCase, startingBitmap *Bitmap) *Bitmap {
+	receiver := NewBitmap()
+	if testCase.useFrozen {
+		frozenBytes, err := startingBitmap.Freeze()
+		require.NoError(t, err)
+		if testCase.useMMapped {
+			require.NoError(t, receiver.FrozenView(getBytesAsMMap(t, frozenBytes)))
+			return receiver
+		}
+		require.NoError(t, receiver.FrozenView(frozenBytes))
+		return receiver
+	}
+	nonFrozenBytes, err := startingBitmap.ToBytes()
+	require.NoError(t, err)
+
+	if testCase.useMMapped {
+		_, err = receiver.FromBuffer(getBytesAsMMap(t, nonFrozenBytes))
+		require.NoError(t, err)
+		return receiver
+	}
+	_, err = receiver.FromBuffer(nonFrozenBytes)
+	require.NoError(t, err)
+	return receiver
+}
+
+func getBytesAsMMap(t *testing.T, startingBytes []byte) []byte {
+	os.WriteFile("data", startingBytes, 0777)
```

Comment:
Surely, there is some needed error handling here?

Note that os.WriteFile is go 1.16. I think you want ioutil.WriteFile for portability.